### PR TITLE
reduce listen socket permissions to 0660

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
because, why not?
#### Does this PR introduce a user-facing change?

```release-note
Reduce the permission on the listen socket to 0660
```

### CODE REVIEW

**Comment by fidencio** [comment]
LGTM, thanks @haircommander!

If there's a bug report about this, please, link to the bug in the commit message.

**Review by saschagrunert — APPROVED** [review]
Just a nit, otherwise LGTM

File: cmd/crio/main.go [hunk 1]
```diff
@@ -214,6 +214,10 @@ func main() {
 			logrus.Fatalf("failed to listen: %v", err)
 		}
 
+		if err := os.Chmod(config.Listen, 0o660); err != nil {
+			logrus.Fatalf("failed to chmod listen socket %s: %v", config.Listen, err)
```
[the remark this answers is not in the document]
Comment (haircommander) [reply cmd/crio/main.go]: fixed that and another case

**Comment by haircommander** [comment]
oops I had titled this wrong

my intuition is that 660 is good for now, so that users can run crictl in the root group but not as root

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**